src/hooks/useResizeObserver.ts

```diff
@@ -0,0 +1,37 @@
+import { RefObject, useEffect, useState } from "react";
+
+interface Options {
+  box: "content-box" | "border-box" | "device-pixel-content-box";
+}
+
+/**
+ *
+ * useResizeObserver hook
+ *
+ * Returns a resize observer for a React Ref and fires a callback
+ * https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver
+ *
+ * @param {RefObject<HTMLElement | null>} ref React ref on which resizes are to be observed
+ * @param {ResizeObserverCallback} callback Function that needs to be fired on resize
+ * @param {Options} options An options object allowing you to set options for the observation
+ */
+function useResizeObserver(
```

Comment:
I think I want to get away from the pattern of using `mutableRef`s here and instead use `Callback` refs. Also let's change this hook to be `useResizeObserverRef` and create a ref in the hook and then expose it out. 

```jsx
const [ref] = useResizeObserver(callback, options)

<div ref={ref}>

</div>
```

Reply:
Main reason for it being that `mutableRefs` don't really track node changes. Apologies for not mentioning this in advance but this is a better solution in the long run and I would appreciate this change a lot! @qiweiii